### PR TITLE
Add examples app with bundled react vanilla, material examples

### PR DESCRIPTION
**Pull request by lucas-koehler**
This introduces a new private package `examples-app` to create a simple web page that ultimately has child pages for every example.

In this initial commit, only the react vanilla and material examples are added. Therefore, they are bundled with rollup to be able to run as standalone pages.

- Add rollup configs to bundle react vanilla and material examples
- Add required dev deps and upgrade rollup dev dep tp ^2.78.0 for all packages
  - **NOTE**: The update was only necessary for the two changed packages. However, I updated the dep everywhere to keep it consistent
- change material deps to ~5.2.2 because latest 5.x versions break api for Autocomplete
- Add private package `examples-app` with a script to aggregate the examples
- Add root level npm script to create the examples app
- regenerate package-lock.json

Part of #1706

### Netlify instructions

Build examples app:
```
npm ci && npm run init && npm run examples-app
```

App will be located in `packages/examples-app/dist/`

### CODE REVIEW

**Comment by sdirix** [comment]
@lucas-koehler can you resolve the conflicts?

**Comment by lucas-koehler** [comment]
@sdirix I resolved the conflict.

File: packages/examples-app/prepare-examples-app.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = dirname(__filename);
+
+const distDir = join(__dirname, 'dist');
+const packagesDir = join(__dirname, '..');
+const examples = {
+  'react-vanilla': join(packagesDir, 'vanilla', 'example', 'dist'),
```
Comment (sdirix) [inline packages/examples-app/prepare-examples-app.js:14]:
```suggestion
  'react-vanilla': join(packagesDir, 'vanilla-renderers', 'example', 'dist'),
```

**Comment by lucas-koehler** [comment]
@sdirix Thanks for the review and suggestions. I applied the suggestions and renamed the scripts as suggested. A brief retest did not show any problems :)

**Review by sdirix — APPROVED** [review]
Works for me

**Pull request merged**